smoke-tests/src/test/java/com/splunk/opentelemetry/TelemetryRetriever.java

```diff
@@ -0,0 +1,118 @@
+import io.opentelemetry.proto.collector.trace.v1.ExportTraceServiceRequest;
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.ResponseBody;
+
+class TelemetryRetriever {
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+  private final OkHttpClient client;
+  private final int backendPort;
+
+  TelemetryRetriever(OkHttpClient client, int backendPort) {
+    this.client = client;
+    this.backendPort = backendPort;
+  }
+
+  void clearTelemetry() throws IOException {
+    client
+        .newCall(
+            new Request.Builder()
+                .url(String.format("http://localhost:%d/clear", backendPort))
+                .build())
+        .execute()
+        .close();
+  }
+
+  TraceInspector waitForTraces() throws IOException, InterruptedException {
+    Stream<JsonNode> content = waitForContent("get-traces");
+
+    return new TraceInspector(
+        content.map(this::deserializeTraceRequest).collect(Collectors.toList()));
+  }
+
+  MetricsInspector waitForMetrics() throws IOException, InterruptedException {
+    Stream<JsonNode> content = waitForContent("get-metrics");
+
+    return new MetricsInspector(
+        content.map(this::deserializeMetricsRequest).collect(Collectors.toList()));
+  }
+
+  private ExportTraceServiceRequest deserializeTraceRequest(JsonNode it) {
+    var builder = ExportTraceServiceRequest.newBuilder();
+    deserializeIntoBuilder(it, builder);
+    return builder.build();
+  }
+
+  private ExportMetricsServiceRequest deserializeMetricsRequest(JsonNode it) {
+    var builder = ExportMetricsServiceRequest.newBuilder();
+    deserializeIntoBuilder(it, builder);
+    return builder.build();
+  }
+
+  private void deserializeIntoBuilder(JsonNode it, GeneratedMessageV3.Builder<?> builder) {
+    // TODO(anuraaga): Register parser into object mapper to avoid de -> re ->
+    // deserialize.
+    try {
+      JsonFormat.parser().merge(OBJECT_MAPPER.writeValueAsString(it), builder);
+    } catch (InvalidProtocolBufferException | JsonProcessingException e) {
+      throw new IllegalArgumentException("Invalid telemetry data", e);
+    }
+  }
+
+  private Stream<JsonNode> waitForContent(String path) throws IOException, InterruptedException {
+    long previousSize = 0;
+    long deadline = System.currentTimeMillis() + TimeUnit.SECONDS.toMillis(30);
+    String content = "[]";
+    while (System.currentTimeMillis() < deadline) {
+
+      Request request =
+          new Request.Builder()
+              .url(String.format("http://localhost:%d/%s", backendPort, path))
+              .build();
+
+      try (ResponseBody body = client.newCall(request).execute().body()) {
+        content = body.string();
+      }
+
+      if (content.length() > 2 && content.length() == previousSize) {
```

Comment:
This length > 2 isn't instantly clear to me.  Maybe a comment to help?

Reply:
I've added an `EMPTY_CONTENT` constant that hopefully clears that up.